polygon/heimdall/range_index.go

```diff
@@ -0,0 +1,93 @@
+package heimdall
+
+import (
+	"context"
+	"encoding/binary"
+
+	"github.com/c2h5oh/datasize"
+	"github.com/ledgerwatch/log/v3"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon-lib/kv/mdbx"
+)
+
+type RangeIndex struct {
+	db kv.RwDB
+}
+
+const rangeIndexTableName = "Index"
```

Comment:
so this `RangeIndex` struct - is it meant to be re-used for different types of range data like Milestones and Events in addition to Checkpoints? If yes then should the table names be a bit more specific so that we have separate tables for each index? Maybe a `kv.TableCfg` should be passing in as an input to `NewRangeIndex`?

Reply:
Currently each index a separate in-memory object implemented as an mdbx in-memory db. I prefer mdbx stay as a private implementation detail.

Reply:
If that is the case can you not pass `tmpdir` to the constructor?
have you tested this works when you try to create 3 "in-mem" mdmx instances with the same "tmpdir"?